src/video/opengl.cpp

```diff
@@ -179,6 +222,42 @@ bool IsOpenGLVersionAtLeast(byte major, byte minor)
 	return (_gl_major_ver > major) || (_gl_major_ver == major && _gl_minor_ver >= minor);
 }
 
+/** Bind basic information functions. */
+static bool BindBasicInfoProcs()
+{
+	_glGetString = (PFNGLGETSTRINGPROC)GetOGLProcAddress("glGetString");
+	_glGetIntegerv = (PFNGLGETINTEGERVPROC)GetOGLProcAddress("glGetIntegerv");
+	_glGetError = (PFNGLGETERRORPROC)GetOGLProcAddress("glGetError");
+
+	return _glGetString != nullptr && _glGetIntegerv != nullptr && _glGetError != nullptr;
+}
+
+/** Bind OpenGL 1.0 and 1.1 functions. */
+static bool BindBasicOpenGLProcs()
+{
+	_glDisable = (PFNGLDISABLEPROC)GetOGLProcAddress("glDisable");
+	_glEnable = (PFNGLENABLEPROC)GetOGLProcAddress("glEnable");
+	_glViewport = (PFNGLVIEWPORTPROC)GetOGLProcAddress("glViewport");
+	_glTexImage1D = (PFNGLTEXIMAGE1DPROC)GetOGLProcAddress("glTexImage1D");
+	_glTexImage2D = (PFNGLTEXIMAGE2DPROC)GetOGLProcAddress("glTexImage2D");
+	_glTexParameteri = (PFNGLTEXPARAMETERIPROC)GetOGLProcAddress("glTexParameteri");
+	_glTexSubImage1D = (PFNGLTEXSUBIMAGE1DPROC)GetOGLProcAddress("glTexSubImage1D");
+	_glTexSubImage2D = (PFNGLTEXSUBIMAGE2DPROC)GetOGLProcAddress("glTexSubImage2D");
+	_glBindTexture = (PFNGLBINDTEXTUREPROC)GetOGLProcAddress("glBindTexture");
+	_glDeleteTextures = (PFNGLDELETETEXTURESPROC)GetOGLProcAddress("glDeleteTextures");
+	_glGenTextures = (PFNGLGENTEXTURESPROC)GetOGLProcAddress("glGenTextures");
+	_glPixelStorei = (PFNGLPIXELSTOREIPROC)GetOGLProcAddress("glPixelStorei");
+	_glClear = (PFNGLCLEARPROC)GetOGLProcAddress("glClear");
+	_glClearColor = (PFNGLCLEARCOLORPROC)GetOGLProcAddress("glClearColor");
+	_glBlendFunc = (PFNGLBLENDFUNCPROC)GetOGLProcAddress("glBlendFunc");
+	_glDrawArrays = (PFNGLDRAWARRAYSPROC)GetOGLProcAddress("glDrawArrays");
+
+	return _glDisable != nullptr && _glEnable != nullptr && _glViewport != nullptr && _glTexImage1D != nullptr && _glTexImage2D != nullptr &&
+		_glTexParameteri != nullptr && _glTexSubImage1D != nullptr && _glTexSubImage2D != nullptr && _glBindTexture != nullptr &&
+		_glDeleteTextures != nullptr && _glGenTextures != nullptr && _glPixelStorei != nullptr && _glClear != nullptr && _glClearColor != nullptr &&
+		_glBlendFunc != nullptr && _glDrawArrays != nullptr;
```

Comment:
This is asking for mistakes :D Any way to do this nicer? (honest question).

`if (!(_glDisable = (PFNGLDISABLEPROC)GetOGLProcAddress("glDisable")) return false`

is frowned upon I guess :D

Reply:
IDK, I feel like I'd prefer your version - the duplication and possibility of missing a function out is worse in my eyes. Perhaps a macro could be used to combine the two somehow?